lib/kinesis_client.rb

```diff
@@ -96,7 +90,21 @@ def push_batch(batch)
   end
 
   def push_records
-    @batch.each_slice(@batch_size) {|slice| push_batch slice}
-    @batch = []
+    if @records.length > 0 && @records.length < @batch_size
+      push_batch(@records)
+    else
+      @records.each_slice(@batch_size) { |slice| push_batch(slice) }
+    end
+    @records = []
+  end
```

Comment:
1. I don't think we want to push a batch if  `records.length` is 0. 
2. We'll want to add some tests for the method now that it has changed. We can talk about what those would look like if you want.